### PR TITLE
Adding port-naming-validation for service with sidecar

### PR DESCRIPTION
**Pull request by xeviknal**
** Describe the change **
Don't run the port name pattern validation for services out of the service mesh (without sidecar).

** Issue reference **
https://github.com/kiali/kiali/issues/1891

### CODE REVIEW

**Comment by lucasponce** [comment]
Travis is complaining about format but logic seem ok to me.

**Comment by hhovsepy** [comment]
I still can see validations:
Screenshot from 2019-11-22 15-25-52

`apiVersion:v1
kind: Service
metadata:
  name: ratings-java-svc
  namespace: ratings-java
  labels:
    app: ratings
    service: ratings-svc
spec:
  ports:
  - port: 9080
    name: http
  selector:
    app: ratings-java
    version: v1
---
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: ratings-java
  namespace: ratings-java
  labels:
    app: ratings-java
    version: v1
spec:
  replicas: 1
  template:
    metadata:
      annotations:
         sidecar.istio.io/inject: "false"
      labels:
        app: ratings-java
        version: v1
    spec:
      containers:
      - name: ratings-java
        image: pilhuhn/ratings-java:f
        imagePullPolicy: IfNotPresent
        ports:
        - containerPort: 8080`

**Comment by xeviknal** [comment]
@hhovsepy I think the validation you shown is not because of the port naming. It is because of mismatch of the port between service and deployment. Can you confirm that?
Also see this comment: https://github.com/kiali/kiali/issues/1891#issuecomment-557519775
I will be another PR upcoming that will make this be a warning.

**Comment by hhovsepy** [comment]
@xeviknal right, the ports are different

**Comment by lucasponce** [comment]
I'm testing this locally, I hope to have it merged this morning.

File: business/checkers/services/port_mapping_checker.go [hunk 1]
```diff
@@ -4,38 +4,50 @@ import (
...
-	for portIndex, sp := range p.Service.Spec.Ports {
-		if strings.ToLower(string(sp.Protocol)) == "udp" {
-			continue
-		} else if !kubernetes.MatchPortNameWithValidProtocols(sp.Name) {
-			validation := models.Build("port.name.mismatch", fmt.Sprintf("spec/ports[%d]", portIndex))
-			validations = append(validations, &validation)
+	// Check Port naming for services in the service mesh
+	if p.hasMatchingPodsWithSidecar(p.Service) {
+		for portIndex, sp := range p.Service.Spec.Ports {
+			if strings.ToLower(string(sp.Protocol)) == "udp" {
```
Comment (lucasponce) [inline business/checkers/services/port_mapping_checker.go]: Not sure if the "udp" is worth to locate it here or inside the "MatchPortNameWithValidProtocols" logic.
But not a blocker, just thinking loud.

**Comment by hhovsepy** [comment]
Screenshot from 2019-12-02 10-53-01

**Pull request merged**